### ISSUE TITLE
bug: infinite loop is happening

### ISSUE THREAD

**Issue by amzapanta** [Group4-Scheme/scheme-program#9]
So while I was checking Kevin's code, I found a bug in our program. I traced where the problem is coming from and it showed me this code snippet. 

![image](https://github.com/Group4-Scheme/scheme-program/assets/142793396/f2ce013a-d1a5-4a1b-aa34-5c97a09e0c3e)

It is on line 282-287 of our scheme-program.scm file.

I think it has something to do with the do-while loop, it keeps on running and never stopping

**Comment by HansDaligdig** [comment]
The do-loop seems to be running indefinetely. Adz, do you have any specific test cases or scenarios that trigger this behavior?

**Comment by DianaCarino** [comment]
I took a look at the code, and it seems like the termination condition is missing in the loop. Adz, could you share more insights into the expected behavior and any particular use cases where this problem arises?

**Comment by SLU-Dominic** [comment]
> I took a look at the code, and it seems like the termination condition is missing in the loop. Adz, could you share more insights into the expected behavior and any particular use cases where this problem arises?

I agree with @DianaCarino. @amzapanta, understanding the context and expected behavior of the do-loop would be helpful.

**Comment by eligigante** [comment]
I am looking at the code too. I think @DianaCarino might be onto something. The termination condition seems incomplete. I think there should be a utilization of a temp veriable?

**Comment by amzapanta** [comment]
Appreciate the insights, everyone. The termination condition indeed looks like the potential culprit. I'll run some test cases and share the results.

**Comment by amzapanta** [comment]
Hey guys, after some investigation, I found that adding (do ((check x (- check 1))) after (do ((check x)) resolves the issue. It ensures the "check" variable decreases with each iteration, meeting the termination condition. Thoughts?

**Comment by HansDaligdig** [comment]
Great find, Adz! That indeed addresses the termination condition, and the loop should now stop as intended. Good collaboration, everyone!

**Comment by DianaCarino** [comment]
Nice catch, Adz! The additional loop iteration with the decrementing "check" resolves the issue. Appreciate the teamwork here.

**Comment by SLU-Dominic** [comment]
> Hey guys, after some investigation, I found that adding (do ((check x (- check 1))) after (do ((check x)) resolves the issue. It ensures the "check" variable decreases with each iteration, meeting the termination condition. Thoughts?

Well done, @amzapanta. It's a logical solution that improves the code. Thanks for sharing.

**Comment by eligigante** [comment]
Thanks for updating us, @amzapanta! The solution aligns with the intended logic of the do-loop. I think it's okay now to close this issue.
